tss-esapi/src/structures/attestation.rs

```diff
@@ -0,0 +1,292 @@
+    pub fn attestation_type(&self) -> AttestationType {
+        match self {
+            AttestInfo::Certify { .. } => AttestationType::Certify,
+        }
+    }
+}
+
+impl TryFrom<AttestInfo> for TPMU_ATTEST {
+    type Error = Error;
+
+    fn try_from(native: AttestInfo) -> Result<Self> {
+        Ok(match native {
+            AttestInfo::Certify {
+                name,
+                qualified_name,
+            } => TPMU_ATTEST {
+                certify: TPMS_CERTIFY_INFO {
+                    name: name.try_into()?,
+                    qualifiedName: qualified_name.try_into()?,
+                },
+            },
+        })
+    }
+}
+
+/// Type of attestation.
+///
+/// # Details
+/// Corresponds to `TPMI_ST_ATTEST`.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum AttestationType {
+    Certify,
+    Quote,
+    SessionAudit,
+    CommandAudit,
+    Time,
+    Creation,
+    Nv,
+    NvDigest,
+}
+
+impl From<AttestationType> for TPMI_ST_ATTEST {
+    fn from(native: AttestationType) -> Self {
+        match native {
+            AttestationType::Certify => TPM2_ST_ATTEST_CERTIFY,
+            AttestationType::Quote => TPM2_ST_ATTEST_QUOTE,
+            AttestationType::SessionAudit => TPM2_ST_ATTEST_SESSION_AUDIT,
+            AttestationType::CommandAudit => TPM2_ST_ATTEST_COMMAND_AUDIT,
+            AttestationType::Time => TPM2_ST_ATTEST_TIME,
+            AttestationType::Creation => TPM2_ST_ATTEST_CREATION,
+            AttestationType::Nv => TPM2_ST_ATTEST_NV,
+            AttestationType::NvDigest => TPM2_ST_ATTEST_NV_DIGEST,
+        }
+    }
+}
+
+impl TryFrom<TPMI_ST_ATTEST> for AttestationType {
+    type Error = Error;
+
+    fn try_from(tss: TPMI_ST_ATTEST) -> Result<Self> {
+        match tss {
+            TPM2_ST_ATTEST_CERTIFY => Ok(AttestationType::Certify),
+            TPM2_ST_ATTEST_QUOTE => Ok(AttestationType::Quote),
+            TPM2_ST_ATTEST_TIME => Ok(AttestationType::Time),
+            TPM2_ST_ATTEST_CREATION => Ok(AttestationType::Creation),
+            TPM2_ST_ATTEST_NV => Ok(AttestationType::Nv),
+            TPM2_ST_ATTEST_NV_DIGEST => Ok(AttestationType::NvDigest),
+            TPM2_ST_ATTEST_SESSION_AUDIT => Ok(AttestationType::SessionAudit),
+            TPM2_ST_ATTEST_COMMAND_AUDIT => Ok(AttestationType::CommandAudit),
+            _ => Err(Error::local_error(WrapperErrorKind::InvalidParam)),
+        }
+    }
+}
+
+/// Attestation data buffer.
+///
+/// # Details
+/// Corresponds to `TPM2B_ATTEST`. The contents of
+/// the buffer can be unmarshalled into an [Attest]
+/// structure.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct AttestBuffer(Zeroizing<Vec<u8>>);
+
+impl Default for AttestBuffer {
+    fn default() -> Self {
+        AttestBuffer(Vec::new().into())
+    }
+}
+
+impl AttestBuffer {
+    pub const MAX_SIZE: usize = 2304;
+
+    pub fn value(&self) -> &[u8] {
+        &self.0
+    }
+}
+
+impl Deref for AttestBuffer {
+    type Target = Vec<u8>;
```

Comment:
Maybe a bit more flexible to set `Target = [u8]`? That way, we return a `&[u8]`, which is more flexible than a `&Vec<u8>` (and fits with the `Vec` deref: https://doc.rust-lang.org/std/vec/struct.Vec.html#impl-Deref).